lightning_toolbox/data/module.py

```diff
@@ -104,7 +104,10 @@ def __init__(
         self.train_num_workers = train_num_workers if train_num_workers is not None else num_workers
         self.val_num_workers = val_num_workers if val_num_workers is not None else num_workers
         self.test_num_workers = test_num_workers if test_num_workers is not None else num_workers
-
+        
+        # Save all the hyper parameters for further reference:
+        self.save_hyperparameters()
```

Comment:
Please first make sure that this does not interfere with calls to `lightning.LitningModule.save_hyperparameters`. Does a call to the modul's save_hparams function, overwrite the hparams saved by the datamodule? If so what is the call pipeline, is datamodule instantiated first, and then the lightning module? 
If you are sure that these two don't have conflicts, please also add a `save_hparams: bool = True` to the datamodule constructor(look at [`lightning_toolbox.TrainingModule:53`](https://github.com/vahidzee/lightning-toolbox/blob/69f9dcceb9a5a10ecf8012b496391ef61f3b2d38/lightning_toolbox/training/module.py#L53)). This is because, if someone inherits the Datamodule class, they might want to change the parameters passed on to the save_hyperparameters function call. This function although undocumented, looks at the call context of the caller, which by default is the datamodule here, and saves all the existing variables as hyper parameters, through intercepting python interpreters global context.
You can imagine that when you inherit from this datamodule, you might want to save other parameters that might be meaningful to your own usecase, or even chose to ignore some of the parameters that are present in `lightning_toolbox.DataModule` for tidy keeping.

Reply:
Now that you're on it also add a `initialize_superclass: bool = True,` to the constructor just like [here](https://github.com/vahidzee/lightning-toolbox/blob/69f9dcceb9a5a10ecf8012b496391ef61f3b2d38/lightning_toolbox/training/module.py#L56).